### PR TITLE
[4.0] save/cancel button icon color

### PR DESCRIPTION
**Pull request by N6REJ**
Pull Request for Issue #29989 

### Summary of Changes
Changes icons to white on save/cancel buttons.

### Testing Instructions
apply pr
run npm ci or download prebuilt package.
go to administrator/index.php?option=com_content&view=article&layout=edit&id=2
click on "images & links"
verify save/cancel icons are white
install "blog sample data"
log into front end as super admin.
go to index.php?option=com_config&view=templates&Itemid=112
verify save/cancel buttons have white icons

### Actual result BEFORE applying this Pull Request
image
image

### Expected result AFTER applying this Pull Request
image

image

### Documentation Changes Required
none

### CODE REVIEW

File: administrator/templates/atum/scss/vendor/bootstrap/_form.scss [hunk 1]
```diff
@@ -73,6 +73,13 @@ select.form-control {
...
   }
 
+  .button-clear .fa-times {
+    color: white;
+    opacity: .9;
+    }
+
+
+
   @include media-breakpoint-down(md) {
```
Comment (Quy) [inline administrator/templates/atum/scss/vendor/bootstrap/_form.scss]:
```suggestion
  }

  @include media-breakpoint-down(md) {
```

**Comment by infograf768** [comment]
This is not enough for article edit (frontend):

Screen Shot 2020-08-05 at 13 10 15

**Comment by infograf768** [comment]
It is also not enough for module edit (frontend)

Screen Shot 2020-08-05 at 13 16 23

Same for site settings

Screen Shot 2020-08-05 at 13 21 39

**Comment by N6REJ** [comment]
in variables.scss how can I tell it that $success and $danger are default values only.
compiler doesn't like !default
```scss
$theme-colors: map-merge((
  primary:                         $base-color, //used only in bootstrap, please use $atum-bg-dark
  secondary:                       $gray-700, //used for btn-secondary
  success:                         $green,
  info:                            $light-blue,
  warning:                         $yellow,
  danger:                          $red,
  light:                           $gray-100, //used in bootstrap
  dark:                            $gray-900,  //used in bootstrap and for table-border
  atum-link-color:                 $light-blue,
  atum-text-dark:                  $gray-700,
  atum-bg-dark:                    $base-color, //standard background color
  action:                          $base-color,
  error:                           $red-dark,
  alert-success:                   $green-dark
), $theme-colors);
```

the problem is being caused by this overriding form.scss
```scss
.icon-publish,
.fa-check {
  color: theme-color("success");

  .dropdown-status-group & {
    color: inherit;
  }
}

.icon-unpublish,
.fa-times {
  color: theme-color("danger");

  .dropdown-status-group & {
    color: inherit;
  }
}
```

**Comment by infograf768** [comment]
Why not just deleting these classes in `/templates/cassiopeia/scss/blocks/_icons.scss` ?
tested here and looks fine.

**Comment by N6REJ** [comment]
> Why not just deleting these classes in `/templates/cassiopeia/scss/blocks/_icons.scss` ?
> tested here and looks fine.

I'll go hang my head in shame now... I guess I'm so used to things being in J! for reasons I don't understand that that thought never crossed my mind.

**Comment by N6REJ** [comment]
@infograf768 I think it would be more consistent to go ahead and use variables so that it can be easily changed ( like by the casseopia group ) as needed.

**Comment by infograf768** [comment]
Needs scss correction https://ci.joomla.org/joomla/joomla-cms/34526/1/24

**Comment by chmst** [comment]
@helvecio - frontend template design?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30278">issues.joomla.org/tracker/joomla-cms/30278</a>.</sub>

**Comment by N6REJ** [comment]
@infograf768 I don't understand what needs to be fixed.

File: administrator/templates/atum/scss/vendor/bootstrap/_form.scss [hunk 2]
```diff
@@ -73,6 +73,11 @@ select.form-control {
     background-color: darken(theme-color("danger"), 8%);
   }
 
+  .button-clear > .fa-check,
+  .button-select > .fa-times {
+    color: $gray-100;
+    }
```
Comment (infograf768) [inline administrator/templates/atum/scss/vendor/bootstrap/_form.scss]: Indent 2 spaces instead of 4

Comment (N6REJ) [reply administrator/templates/atum/scss/vendor/bootstrap/_form.scss]: ah ty.

**Comment by N6REJ** [comment]
@infograf768 it was following the 2 space rule because its inside another {} but I changed storm to 
image
and ran reformatter.
Couldn't change it any other way.

**Comment by infograf768** [comment]
Sorry to say but you made it worse

**Comment by infograf768** [comment]
Here is a corrected one.

[form.txt](https://github.com/joomla/joomla-cms/files/5046809/form.txt)

**Comment by N6REJ** [comment]
@infograf768 according to storm nothing was changed but I applied/saved/pushed it anyway.

Maybe someday I can actually get someone to help me setup storm properly.  I've gotten tired of asking.

**Comment by infograf768** [comment]
Forget storm please. I modified the file for you. This will trigger drone again.

FYI, this is what drone error displayed for scss before I corrected your file.
Screen Shot 2020-08-09 at 10 36 10

**Comment by infograf768** [comment]
And this is what it displays after
Screen Shot 2020-08-09 at 10 45 52

**Comment by N6REJ** [comment]
@infograf768 what ide if any are you using?

**Comment by infograf768** [comment]
I use eclipse, but it is not set to show indentations errors. I just code manually.

**Comment by N6REJ** [comment]
ok, I'm doing my best is all I can say.  Glad to see drone is happy now.. ty for the corrected file.

**Comment by infograf768** [comment]
@N6REJ 

I suggest you change back your phpstorm to the original settings for tabs as it would kill further PRs. There was only one change to do manually before you modified.

Now, remains one change to do for `Save & Close` when editing modules as the icon is white on white.

We get
Screen Shot 2020-08-09 at 11 28 19

I guess we should have
Screen Shot 2020-08-09 at 11 28 41

i.e.
change
```
			<button type="button" class="btn btn-secondary" data-submit-task="modules.save">
				<span class="fas fa-check" aria-hidden="true"></span>
				<?php echo Text::_('JSAVE'); ?>
			</button>
```
to
```
			<button type="button" class="btn btn-primary" data-submit-task="modules.save">
				<span class="fas fa-check" aria-hidden="true"></span>
				<?php echo Text::_('JSAVE'); ?>
			</button>
```

**Comment by sandewt** [comment]
>install "blog sample data"

@N6REJ How to install this?

Note: I have tested the following link in the fontend successfully: ```/index.php/test?view=form&layout=edit&a_id=1```<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30278">issues.joomla.org/tracker/joomla-cms/30278</a>.</sub>

**Comment by infograf768** [comment]
@sandewt
Clean joomla installation.
Then
Screen Shot 2020-08-09 at 12 02 57

patch and run npm ci, display various stuff in frontend

**Comment by sandewt** [comment]
@infograf768  thanks

**Comment by sandewt** [comment]
I have tested this item :white_check_mark: successfully on b97b9cf3323dc90306b0becdeef889cb7e18a96d<br /><br />Joomla! 4.0.0-beta4-dev+pr.30278 Development [ Mañana ] 29-July-2020 18:21 GMT<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30278">issues.joomla.org/tracker/joomla-cms/30278</a>.</sub>

**Comment by infograf768** [comment]
@sandewt 
can’t be ok until https://github.com/joomla/joomla-cms/pull/30278#issuecomment-671030210 is solved. 😉

**Comment by sandewt** [comment]
@infograf768  

I'm waiting :smiley:

**Comment by N6REJ** [comment]
@infograf768 is this correct for scss?
image

php
image

**Comment by N6REJ** [comment]
> There was only one change to do manually before you modified.

Thats just it, it wouldn't do it.  I would hit enter and it would put the 4, which was 2+2

**Comment by N6REJ** [comment]
@infograf768 I made the change you suggested but not sure how to test as I can't find the area you showed the error in.

**Comment by infograf768** [comment]
@N6REJ 
> @infograf768 I made the change you suggested but not sure how to test as I can't find the area you showed the error in.

Frontend module editing.

**Comment by infograf768** [comment]
@sandewt 
now you can test fully.

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 0a47873d904c881fde20718f915570b49b446f51<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30278">issues.joomla.org/tracker/joomla-cms/30278</a>.</sub>

**Comment by sandewt** [comment]
I have tested this item :white_check_mark: successfully on 0a47873d904c881fde20718f915570b49b446f51<br /><br />Joomla! 4.0.0-beta4-dev Development [ Mañana ] 29-July-2020 18:21 GMT

Joomla_4.0.0-beta4-dev+pr.30278-Development-Update_Package.zip

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30278">issues.joomla.org/tracker/joomla-cms/30278</a>.</sub>

**Comment by infograf768** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30278">issues.joomla.org/tracker/joomla-cms/30278</a>.</sub>

**Comment by sandewt** [comment]
Images by test results:

**Comment by infograf768** [comment]
Tks

**Pull request merged**